src/main/java/sparksoniq/jsoniq/item/DurationItem.java

```diff
@@ -119,7 +119,9 @@ public boolean equals(Object otherObject) {
 
     @Override
     public int hashCode() {
-        return this.getValue().hashCode();
+        return String.valueOf(
+            this.getValue().toDurationFrom(Instant.now()).getMillis()
+        ).hashCode();
```

Comment:
I cannot just call _this.getValue().hashCode()_ because the groupBy will not work effectively across Duration and its subtypes. 
I can avoid the cast to String and call the hashCode() on the long value of millis. This should be faster. Thank you

Reply:
Thank, this reasoning makes sense. Good optimization on using Long.hashCode().